caffe2/proto/torch.proto

```diff
@@ -4,547 +4,85 @@ import "caffe2/proto/caffe2.proto";
-  IR_VERSION_2017_10_30 = 0x0000000000000002;
-
-  // IR VERSION 3 published on Nov 3, 2017
-  // - For operator versioning:
-  //    - Added new message OperatorSetIdProto
-  //    - Added opset_import in ModelProto
-  // - For vendor extensions, added domain in NodeProto
-  IR_VERSION_NEWEST_ONNX = 0x0000000000000003;
-
-  // PYTORCH IR VERSION
-  IR_VERSION_NEWEST = 0x0000000000000103;
+  IR_VERSION_NEWEST = 0x0000000000000101;
 }
 
-// Attributes
-//
-// A named attribute containing either singular float, integer, string, graph,
-// and tensor values, or repeated float, integer, string, graph, and tensor values.
-// An AttributeProto MUST contain the name field, and *only one* of the
-// following content fields, effectively enforcing a C/C++ union equivalent.
-message AttributeProto {
-
-  // Note: this enum is structurally identical to the OpSchema::AttrType
-  // enum defined in schema.h.  If you rev one, you likely need to rev the other.
-  enum AttributeType {
-    UNDEFINED = 0;
-    FLOAT = 1;
-    INT = 2;
-    STRING = 3;
-    TENSOR = 4;
-    GRAPH = 5;
-
-    FLOATS = 6;
-    INTS = 7;
-    STRINGS = 8;
-    TENSORS = 9;
-    GRAPHS = 10;
+message MethodDef {
+  // method name
+  optional string name = 1; // method name
+
+  // static graph
+  optional caffe2.NetDef graph = 2;
+  // method is represented as torch script
+  optional string torch_script = 3;
+
+  // the names of inputs and outputs
+  repeated string inputs = 4;
+  repeated string outputs = 5;
+
+  // method type, can be one of the following:
+  //   1) MAIN, the main method, by default is `forward` method
+  //   2) INIT, only run once for initialization
+  //   3) REGULAR, may run multiple times at runtime
+  enum MethodType {
+    MAIN = 1;
+    INIT = 2;
+    REGULAR = 3;
   }
 
-  // The name field MUST be present for this version of the IR.
-  optional string name = 1;           // namespace Attribute
-
-  // if ref_attr_name is not empty, ref_attr_name is the attribute name in parent function.
-  // In this case, this AttributeProto does not contain data, and it's a reference of attribute
-  // in parent scope.
-  // NOTE: This should ONLY be used in function (sub-graph). It's invalid to be used in main graph.
-  optional string ref_attr_name = 21;
-
-  // A human-readable documentation for this attribute. Markdown is allowed.
-  optional string doc_string = 13;
+  optional MethodType method_type = 6;
 
-  // The type field MUST be present for this version of the IR.
-  // For 0.0.1 versions of the IR, this field was not defined, and
-  // implementations needed to use has_field hueristics to determine
-  // which value field was in use.  For IR_VERSION 0.0.2 or later, this
-  // field MUST be set and match the f|i|s|t|... field in use.  This
-  // change was made to accomodate proto3 implementations.
-  optional AttributeType type = 20;   // discriminator that indicates which field below is in use
+  optional string debug_info = 7;
```

Comment:
What goes in here?

Reply:
Reserved for the debug information. Since NetDef already has debug_info, will remove it from MethodDef.